wgpu-hal/src/gles/egl.rs

```diff
@@ -1363,9 +1385,21 @@ impl crate::Surface for Surface {
                     }
                     (WindowKind::Unknown, Rwh::OhosNdk(handle)) => handle.native_window.as_ptr(),
                     (WindowKind::Wayland, Rwh::Wayland(handle)) => {
-                        let library = &self.wsi.display_owner.as_ref().unwrap().library;
+                        let library = &self
+                            .wsi
+                            .display_owner
+                            .as_ref()
+                            .ok_or(crate::SurfaceError::Other("No WSI display owner"))?
+                            .library;
                         let wl_egl_window_create: libloading::Symbol<WlEglWindowCreateFun> =
-                            unsafe { library.get(c"wl_egl_window_create".to_bytes()) }.unwrap();
+                            unsafe { library.get(c"wl_egl_window_create".to_bytes()) }.map_err(
+                                |err| {
+                                    log::error!("Failed to load `wl_egl_window_create`: {err}");
+                                    crate::SurfaceError::Other(
```

Comment:
logging the error we're reporting anyways seems redundant. Hard to tell what's the criteria for log + error return versus just error return
```suggestion
                                |err| {
                                    crate::SurfaceError::Other(
```